lib/improver/ensemble_calibration/ensemble_calibration.py

```diff
@@ -702,6 +702,44 @@ def compute_initial_guess(
                                   no_of_realizations).tolist())
         return np.array(initial_guess, dtype=np.float32)
 
+    @staticmethod
+    def _filter_non_matching_cubes(historic_forecast, truth):
+        """
+        Provide filtering for the historic forecast and truth to make sure
+        that these contain matching validity times. This ensures that any
+        mismatch between the historic forecasts and truth is dealt with.
+
+        Args:
+            historic_forecast (iris.cube.Cube):
+                Cube of historic forecasts that potentially contains
+                a mismatch compared to the truth.
+            truth (iris.cube.Cube):
+                Cube of truth that potentially contains a mismatch
+                compared to the historic forecasts.
+
+        Returns:
+            (tuple): tuple containing
+                matching_historic_forecasts (iris.cube.Cube):
+                    Cube of historic forecasts where any mismatches with
+                    the truth cube have been removed.
+                matching_truths (iris.cube.Cube):
+                    Cube of truths where any mismatches with
+                    the historic_forecasts cube have been removed.
+
+        """
+        matching_historic_forecasts = iris.cube.CubeList([])
+        matching_truths = iris.cube.CubeList([])
+        for hf_slice in historic_forecast.slices_over("time"):
+            coord_values = (
+                {"time": iris_time_to_datetime(hf_slice.coord("time"))})
+            constr = iris.Constraint(coord_values=coord_values)
+            truth_slice = truth.extract(constr)
+            if truth_slice:
+                matching_historic_forecasts.append(hf_slice)
+                matching_truths.append(truth_slice)
+        return (matching_historic_forecasts.merge_cube(),
```

Comment:
From a suite perspective we need to be careful about what happens if one of these merges fail. I think?

Reply:
I've added the raising of an exception to this function, as you're correct that if the historic forecasts and truths do not match in validity time, then we won't be able to calculate any coefficients.

Reply:
I think the chances of this happening in a suite context is very slim, even in the spin up period, so I'm happy that this raises an error.